### PR TITLE
Add data import file upload endpoints

### PR DESCRIPTION
**Pull request by OleksiiKuzminov**

### CODE REVIEW

File: ramls/file.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "description": "Ping message",
```
Comment (KaterynaSenchenko) [inline ramls/file.json]: Please update the description

File: ramls/uploadDefinition.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "description": "Ping message",
```
Comment (KaterynaSenchenko) [inline ramls/uploadDefinition.json]: Same as in file.json

File: descriptors/ModuleDescriptor-template.json [hunk 3]
```diff
@@ -40,6 +62,46 @@
...
+      "description": "Get Upload Definition"
+    },
+    {
+      "permissionName": "data-import.upload.definition.put",
+      "displayName": "Data Import - update upload definition",
+      "description": "Put Upload Definition"
+    },
+    {
+      "permissionName": "data-import.upload.file.post",
+      "displayName": "Data Import - upload file to the storage",
```
Comment (Igor-Gorchakov) [inline descriptors/ModuleDescriptor-template.json]: INto the storage

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -40,6 +62,46 @@
...
+      "description": "upload file"
+    },
+    {
+      "permissionName": "data-import.upload.file.delete",
+      "displayName": "Data Import - delete file from upload definition",
```
Comment (Igor-Gorchakov) [inline descriptors/ModuleDescriptor-template.json]: delete file from the storage ?

File: ramls/file.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  "additionalProperties": false,
+  "properties": {
+    "id": {
+      "description": "UUID",
+      "type": "string"
+    },
+    "location": {
```
Comment (Igor-Gorchakov) [inline ramls/file.json]: maybe "sourcePath" ?

File: ramls/file.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      "description": "The path to the file location",
+      "type": "string"
+    },
+    "name": {
+      "description": "Name of the file with extension",
+      "type": "string"
+    },
+    "jobExecutionId": {
+      "description": "Link to job execution for this file",
```
Comment (Igor-Gorchakov) [inline ramls/file.json]: Link to JobExecution entity

File: ramls/file.json [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      "type": "string"
+    },
+    "uploadDefinitionId": {
+      "description": "Link to Upload Definition",
```
Comment (Igor-Gorchakov) [inline ramls/file.json]: Link to UploadDefinitionEntity

File: src/test/java/org/folio/rest/RestVerticleTest.java [hunk 8]
```diff
@@ -1,68 +1,153 @@
...
   }
 
-  private void sendRequest(String url, HttpMethod method, Handler<HttpClientResponse> handler, String content) {
-    Buffer buffer = Buffer.buffer(content);
-    vertx.createHttpClient()
-      .requestAbs(method, url, handler)
-      .putHeader("x-okapi-tenant", "diku")
-      .putHeader("Accept", "application/json,text/plain")
-      .end(buffer);
+  //@Test
```
Comment (Igor-Gorchakov) [inline src/test/java/org/folio/rest/RestVerticleTest.java]: add TODO if @Test annotation is not necessary for now.

File: src/main/resources/templates/db_scripts/schema.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
+{
+  "tables": [
+    {
+      "tableName": "uploadDefinition",
+      "withMetadata": true,
+      "pkColumnName": "_id",
+      "generateId": false,
+      "withAuditing": false,
```
Comment (Igor-Gorchakov) [inline src/main/resources/templates/db_scripts/schema.json]: maybe better "auditable" ?

File: ramls/uploadDefinition.json [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+      "description": "Status of upload",
+      "type": "string",
+      "enum": ["NEW", "IN_PROGRESS", "LOADED", "COMPLETED"]
+    },
+    "createDate": {
+      "description": "Date and time when the upload definition was created",
+      "type": "string",
+      "format": "date-time"
+    },
+    "files": {
```
Comment (KaterynaSenchenko) [inline ramls/uploadDefinition.json]: Please add description

File: ramls/uploadDefinitionCollection.json [hunk 11]
```diff
@@ -1,21 +1,21 @@
 {
   "$schema": "http://json-schema.org/draft-04/schema#",
   "type": "object",
-  "description": "Collection of ping messages",
+  "description": "Collection of upload definitions",
   "properties": {
-    "pingMessages": {
+    "uploadDefinitions": {
```
Comment (KaterynaSenchenko) [inline ramls/uploadDefinitionCollection.json]: Please add description for the uploadDefinitions property

File: src/test/java/org/folio/rest/RestVerticleTest.java [hunk 12]
```diff
@@ -1,68 +1,153 @@
...
   }
 
-  private void sendRequest(String url, HttpMethod method, Handler<HttpClientResponse> handler, String content) {
-    Buffer buffer = Buffer.buffer(content);
-    vertx.createHttpClient()
-      .requestAbs(method, url, handler)
-      .putHeader("x-okapi-tenant", "diku")
-      .putHeader("Accept", "application/json,text/plain")
-      .end(buffer);
+  //@Test TODO add support of file storing
```
Comment (Igor-Gorchakov) [inline src/test/java/org/folio/rest/RestVerticleTest.java]: +

Comment (Igor-Gorchakov) [inline src/test/java/org/folio/rest/RestVerticleTest.java]: +

**Pull request merged**